doc/esp8266wifi/client-class.rst

```diff
@@ -16,7 +16,21 @@ Methods documented for `Client <https://www.arduino.cc/en/Reference/WiFiClientCo
 9.  `flush() <https://www.arduino.cc/en/Reference/WiFiClientFlush>`__
 10. `stop() <https://www.arduino.cc/en/Reference/WiFIClientStop>`__
 
-Methods and properties described further down are specific to ESP8266. They are not covered in `Arduino WiFi library <https://www.arduino.cc/en/Reference/WiFi>`__ documentation. Before they are fully documented please refer to information below.
+Methods and properties described further down below are specific to ESP8266. Some of them behave differently from the reference `Arduino WiFi library <https://www.arduino.cc/en/Reference/WiFi>`__ , or are only implemented for this Core.
+
+connected
+~~~~~~~~~
+
+Unlike the reference implementation, ``connected()`` means that the client is available for both reads and writes. It will return ``true`` when the client is ``status() == ESTABLISHED`` or ``available() == true``, but will return ``false`` when the ``status() == CLOSED``. Please use ``status()`` for the connection information, ``availableForWrite()`` to check whether it is possible to write, and ``available()`` if you mean to check whether there's unread data.
```

Comment:
> It will return ``true`` when the client is ``status() == ESTABLISHED`` or ``available() == true``, but will return ``false`` when the ``status() == CLOSED``.

According to current code, it will return false if the connection is closed, regardless `available()` result,
meaning "writing is not allowed".

`available()` can still be checked to tell if something has been received and has been kept unread.

https://github.com/esp8266/Arduino/blob/f401f08aba74f6dd0f6dbeb9b1b6b6d94b174b31/libraries/ESP8266WiFi/src/WiFiClient.cpp#L329-L335